apps/daimo-mobile/src/logic/note.ts

```diff
@@ -1,38 +1,21 @@
+import { getNoteClaimSignature } from "@daimo/common";
 import { useEffect, useState } from "react";
-import { Address, Hex, createWalletClient, http, keccak256 } from "viem";
-import { privateKeyToAccount } from "viem/accounts";
-import { baseGoerli } from "viem/chains";
+import { Address, Hex } from "viem";
 
 export function useEphemeralSignature(
-  ephemeralPrivateKey: `0x${string}` | undefined,
-  recipient: Address
+  sender: Address,
+  recipient: Address,
+  ephemeralPrivateKey: `0x${string}` | undefined
 ) {
   const [ephemeralSignature, setEphemeralSignature] = useState<Hex>("0x");
 
   useEffect(() => {
     (async () => {
-      if (!ephemeralPrivateKey) {
-        // Must be sender themselves reclaiming
-        setEphemeralSignature(dummySignature);
-        return;
-      }
-      const ephemeralAccount = privateKeyToAccount(ephemeralPrivateKey);
-      const ephemeralClient = createWalletClient({
-        account: ephemeralAccount,
-        chain: baseGoerli,
-        transport: http(),
-      });
-      const message = keccak256(recipient);
-      const signature = await ephemeralClient.signMessage({
-        message: { raw: message },
-      });
-      setEphemeralSignature(signature);
+      setEphemeralSignature(
+        await getNoteClaimSignature(sender, recipient, ephemeralPrivateKey)
```

Comment:
much nicer!

u might even be able to simplify to

```ts
useEffect(() => {
   getNoteClaimSignature(...).then(setEphemeralSignature);
}, [ephemeralPrivateKey]);
```

Reply:
done